engine/script/src/luacjson/lua_cjson.c

```diff
@@ -684,8 +686,8 @@ static void json_check_encode_depth(lua_State *l, json_config_t *cfg,
     if (current_depth <= cfg->encode_max_depth && lua_checkstack(l, 3))
         return;
 
-    if (!cfg->encode_keep_buffer)
-        strbuf_free(json);
```

Comment:
We should free buffers if error happened - because that's how API designed:
https://github.com/defold/defold/blob/9123aa9aa1aa26953d396ddc8256b8d518fe805e/engine/script/src/script_json.cpp#L154-L160